soc/arm/nxp_imx/mimx8ml8_m7/linker.ld

```diff
@@ -18,3 +18,13 @@ MEMORY
      }
 
 #include <arch/arm/aarch32/cortex_m/scripts/linker.ld>
+
+SECTIONS
+     {
+#ifdef CONFIG_OPENAMP_RSC_TABLE
+        SECTION_PROLOGUE(.resource_table,, SUBALIGN(4))
+        {
+            KEEP(*(.resource_table*))
+        } GROUP_LINK_IN(ROMABLE_REGION)
+#endif
+     }
```

Comment:
does this work with `CONFIG_CMAKE_LINKER_GENERATOR=y`?

Reply:
Although it is possible to build the program with the generated linker script, the resource table section will not be included then.
I used the same mechanism as in the linker script for stm32mp1 - so there is the same behavior.

So if one decides to use a different linker script, for whatever reason, the table is missing, but one will notice that soon.
To save users from this pitfall, maybe CONFIG_OPENAMP_RSC_TABLE could select CONIG_LD_LINKER_TEMPLATE or depend on !CONFIG_CMAKE_LINKER_GENERATOR.
What do you think?

Reply:
> Although it is possible to build the program with the generated linker script, the resource table section will not be included then.

and we should try to keep things aligned, so this is not good.
Today, the `mimx8ml8` does support the CMake linker script generator, try for yourself:
```
cmake -GNinja -DBOARD=mimx8mp_evk_itcm -Bbuild samples/hello_world/ -DCONFIG_CMAKE_LINKER_GENERATOR=y
```

> I used the same mechanism as in the linker script for stm32mp1 - so there is the same behavior.

so that means we now have some work to do:
https://github.com/zephyrproject-rtos/zephyr/issues/43206
but is not a reason for introducing tech-debt in a new PR.

> 
> So if one decides to use a different linker script, for whatever reason, the table is missing, but one will notice that soon. To save users from this pitfall, maybe CONFIG_OPENAMP_RSC_TABLE could select CONIG_LD_LINKER_TEMPLATE or depend on !CONFIG_CMAKE_LINKER_GENERATOR. What do you think?

Linker script generator can also be used for generating ld script 😉 
Is is really that hard to add this line:
```
zephyr_linker_section(NAME resource_table GROUP ROM_REGION NOINPUT)
zephyr_linker_section_configure(SECTION resource_table KEEP INPUT ".resource_table*")
```
here: https://github.com/zephyrproject-rtos/zephyr/blob/9d2d9750b7de4e55c96ace07bb2ae3feabdb3065/soc/arm/nxp_imx/mimx8ml8_m7/CMakeLists.txt

and then test it works ?

Reply:
No, it's not difficult indeed. It was just a lack of knowledge regarding the script generator :/

Thanks to your detailed hint, this section is now included in the output in both linker script variants.
A test in a real system was also successful with both methods.

In both cases, the section will not be added if CONFIG_OPENAMP_RSC_TABLE is not selected.

Thanks for your helpful input!

Reply:
Lack of knoledge also on the script generator...But look strange that we need to duplicate. 

On other side if I suppress one of the declaration either in the linker.ld or the Cmakelists (on stm32mp1 soc), I have `warning: orphan section .resource_table` depending on the CONFIG_CMAKE_LINKER_GENERATOR.

Reply:
> But look strange that we need to duplicate.

this is needed for the time, as the old approach uses the preprocessor to generate the final ld linker script.
The new approach describes everything in CMake, which allows us to then generate linker scripts for different linkers, ld (ld scripts), armlink (scatter files), and more linkers in future.

But it is a transition in progress, and thus needs to be described in both ways for time being.